### ISSUE TITLE
Explain the situation found where listeners run in the opposite of triggering order

### ISSUE THREAD

**Issue by deanrad** [deanrad/polyrhythm#9]
For an event E1, since listeners are not run in parallel, events E2, E3 triggered in the synchronous part of a listener are fully processed (run through filters and listeners) before the next listener for E1 gets to process. Thus forming a stack of events which, when unwound, has listeners for E2/E3 processed in reverse order.

Solution:
- Do state changes in filters only!
- Parallelize the sync part of listeners (seems not possible)

It's worth a reminder that any behavior outside of these invariants are not to be relied upon, and considered non-deterministic!

Invariants:
  - Listeners follow their filters
  - Filters see events in the order they're triggered

```
# With synchronous / non-deferred orderFiller
F:order/place
F:shotgun/fire/client
F:fill/start
                          L:fill/start
F:fill/complete
F:shotgun/fire/server
                          L:fill/complete
                          L:order/place
F:order/pickup
                          L:order/pickup

With asynchronous / deferred orderFiller
F:order/place
F:shotgun/fire/client
                          L:order/place
F:fill/start
                          L:fill/start
F:fill/complete
F:shotgun/fire/server
                          L:fill/complete
F:order/pickup
                          L:order/pickup

# With a changing state in a filter, instead of a listener
F:order/place
                          SF:order/place
F:shotgun/fire/client
F:fill/start
                          SF:fill/start
F:fill/complete
                          SF:fill/complete
F:shotgun/fire/server
F:order/pickup
                          SF:order/pickup

```
In other words - the current behavior is that Listeners can trust that all **Filters** of previous events will have run and completed, but currently cannot trust that all **Listeners** of previous events would have begun.

It might be a stronger guarantee to enqueue the evaluation of all listeners as a microtask (use `microq`?) Errors would still be safe because the triggerer never had sync access to the error.

From [Mountain Dew example app](https://gist.github.com/deanius/bb26a5d31dcd9657772a11ba6b4d1ce6)

**Comment by deanrad** [comment]
This issue surfaced, and was solved here! https://github.com/deanrad/omnibus-rxjs/issues/2
